src/main/java/org/datadog/jmxfetch/RemoteConnection.java

```diff
@@ -53,6 +54,10 @@ public RemoteConnection(LinkedHashMap<String, Object> connectionParams)
             }
 
         }
+        
+        //Set an RMI timeout so we don't get stuck waiting for a bean to report a value
+        System.setProperty("sun.rmi.transport.tcp.responseTimeout", DEFAULT_RMI_RESPONSE_TIMEOUT);
```

Comment:
what happens when we reach the timeout? Do we catch and handle the exception that's raised correctly?

Reply:
An error should be shown during the refresh bean list period where we scope out all available beans. If we hit the timeout there, an exception will be thrown (with a read timeout) and we should wait until the next collection period to try again.